File: app/src/main/java/me/jdvp/androidaspectexample/activity/MainActivity.java
package me.jdvp.androidaspectexample.activity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import dagger.android.support.DaggerAppCompatActivity;
import me.jdvp.androidaspectexample.R;

public class MainActivity extends DaggerAppCompatActivity {
    @BindView(R.id.button_1)
    Button button1;

    @BindView(R.id.button_2)
    Button button2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        //Not a lambda on purpose
        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "Button 1 clicked", Toast.LENGTH_SHORT).show();
            }
        });

        button2.setOnClickListener(v -> Toast.makeText(MainActivity.this, "Button 2 clicked", Toast.LENGTH_SHORT).show());
    }
}

File: app/src/main/java/me/jdvp/androidaspectexample/activity/MainActivity.kt
package me.jdvp.androidaspectexample.activity

import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import me.jdvp.androidaspectexample.R

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //Using findViewById so we don't have to worry about deprecations later
        val button1 = findViewById<Button>(R.id.button_1)
        val button2 = findViewById<Button>(R.id.button_2)

        //The button ClickListener's below only have actions that create Toasts.
        //However, when you press the buttons at runtime, you will notice in LogCat
        //that messages are being logged as well. Those messages are being logged by
        //the Logger class as specified by aspects put in the AspectLogging file. The code
        //to do the logging is injected in-line at compile time

        //Not a lambda on purpose for demonstration purposes
        @Suppress("ObjectLiteralToLambda")
        button1.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                AlertDialog.Builder(this@MainActivity)
                    .setMessage("Button 1 clicked")
                    .show()
            }
        })

        button2.setOnClickListener {
            AlertDialog.Builder(this@MainActivity)
                .setMessage("Button 2 clicked")
                .show()
        }
    }
}